### PR TITLE
Search feature

### PR DESCRIPTION
**Pull request by k1mka**

### CODE REVIEW

File: lib/data/network_services/mocked_network_service.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import 'package:spotify_prototype/data/models/track_model.dart';
+
+abstract class NetworkService {
+  Future<List<TrackModel>> fetchModel({
+    required String search,
+    required int limit,
+  });
+}
+
+class MockedServiceImpl extends NetworkService {
```
Comment (zddarova) [inline lib/data/network_services/mocked_network_service.dart:10]: i would like to see mocked in different file

File: lib/data/network_services/network_service.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import 'dart:convert';
+
+import 'package:http/http.dart';
+import 'package:spotify_prototype/data/models/track_model.dart';
+import 'package:spotify_prototype/data/network_services/mocked_network_service.dart';
+
+class NetworkServiceImpl extends NetworkService {
+  static const tokenValue =
```
Comment (zddarova) [inline lib/data/network_services/network_service.dart:8]: are u planning to implement token getting properly?

File: lib/data/network_services/network_service.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  static const _artistsKey = 'artists';
+
+  @override
+  Future<List<TrackModel>> fetchModel({
+    required String search,
+    required int limit,
+  }) async {
+    final url = Uri.parse('$spotifyUrl$searchEndpoint${searchParamsQuery(search, limit)}');
+
+    final response = await get(url, headers: authHeadersMap);
```
Comment (zddarova) [inline lib/data/network_services/network_service.dart:38]: dont u want to implement separated method for get requests and call it here?

File: lib/presentation/screens/likest_screen/likest_layout.dart [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  const LikestLayout({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: const Text('Likest'),
+          backgroundColor: Colors.green,
+        ),
+        body: const Center(child: Text('List of likest tracks')));
```
Comment (zddarova) [inline lib/presentation/screens/likest_screen/likest_layout.dart]: add trailing commas here?

File: lib/presentation/screens/likest_screen/likest_screen.dart [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import 'package:flutter/material.dart';
+import 'package:spotify_prototype/presentation/screens/likest_screen/likest_layout.dart';
+
+class LikestScreen extends StatelessWidget {
```
Comment (zddarova) [inline lib/presentation/screens/likest_screen/likest_screen.dart:4]: add todo to add bloc here?

File: lib/presentation/screens/playlist_screen/bloc/traclist_bloc.dart [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import 'package:spotify_prototype/data/repository/repository.dart';
+import 'package:spotify_prototype/presentation/screens/playlist_screen/bloc/tracklist_events.dart';
+import 'package:spotify_prototype/presentation/screens/playlist_screen/bloc/tracklist_states.dart';
+
+class PlayListBloc extends Bloc<TrackEvent, PlayListState> {
+  Repository repository;
+  PlayListBloc(this.repository) : super(InitialState()) {
+    on<SearchTrackEvent>((event, emit) async {
+      emit(LoadingTrack());
+      emit(LoadedTrack(await repository.fetchTrack(search: event.search, limit: event.limit)));
```
Comment (zddarova) [inline lib/presentation/screens/playlist_screen/bloc/traclist_bloc.dart:13]: handle errors here?

File: lib/presentation/screens/playlist_screen/bloc/traclist_bloc.dart [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:spotify_prototype/data/repository/repository.dart';
+import 'package:spotify_prototype/presentation/screens/playlist_screen/bloc/tracklist_events.dart';
+import 'package:spotify_prototype/presentation/screens/playlist_screen/bloc/tracklist_states.dart';
+
+class PlayListBloc extends Bloc<TrackEvent, PlayListState> {
+  Repository repository;
```
Comment (zddarova) [inline lib/presentation/screens/playlist_screen/bloc/traclist_bloc.dart:7]: format file pls. Add spaces between lines

File: lib/presentation/widgets/search_widget.dart [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    controller.addListener(() {
+      if (controller.text.isNotEmpty) {
+        context.read<PlayListBloc>().add(SearchTrackEvent(controller.text, 20));
+      }
+    });
+    super.initState();
+    // инит стейт
+  }
+
+  // запрос не в начале
```
Comment (zddarova) [inline lib/presentation/widgets/search_widget.dart]: should be todo
and now it could be removed

**Review by zddarova — APPROVED** [review]

File: lib/data/network_services/network_service.dart (hunk 2 above)
Comment (k1mka) [reply lib/data/network_services/network_service.dart:8]: Yes, I`m planning.

File: lib/data/network_services/network_service.dart (hunk 3 above)
Comment (k1mka) [reply lib/data/network_services/network_service.dart:38]: Yes, I want. But actually, I don't know how to do that.

File: lib/presentation/screens/likest_screen/likest_layout.dart (hunk 4 above)
Comment (k1mka) [reply lib/presentation/screens/likest_screen/likest_layout.dart]: Ok

File: lib/presentation/screens/likest_screen/likest_screen.dart (hunk 5 above)
Comment (k1mka) [reply lib/presentation/screens/likest_screen/likest_screen.dart:4]: Ok

File: lib/presentation/screens/playlist_screen/bloc/traclist_bloc.dart (hunk 6 above)
Comment (k1mka) [reply lib/presentation/screens/playlist_screen/bloc/traclist_bloc.dart:13]: Are you sure what I must do this?

File: lib/presentation/screens/playlist_screen/bloc/traclist_bloc.dart (hunk 7 above)
Comment (k1mka) [reply lib/presentation/screens/playlist_screen/bloc/traclist_bloc.dart:7]: Ok

File: lib/data/network_services/mocked_network_service.dart (hunk 1 above)
Comment (zddarova) [reply lib/data/network_services/mocked_network_service.dart:10]: @k1mka i prefer next structure
network service
mocked network service
network service impl

each class in diffener file

File: lib/data/network_services/network_service.dart (hunk 2 above)
Comment (zddarova) [reply lib/data/network_services/network_service.dart:8]: so add todo here

File: lib/data/network_services/network_service.dart (hunk 3 above)
Comment (zddarova) [reply lib/data/network_services/network_service.dart:38]: add todo at least

File: lib/presentation/screens/playlist_screen/bloc/traclist_bloc.dart (hunk 6 above)
Comment (zddarova) [reply lib/presentation/screens/playlist_screen/bloc/traclist_bloc.dart:13]: ofc u are. U have state, but u are not using it. And main reason - it is bad to show user loading indicator when u will not load anything coz of error.

**Pull request merged**